tests/test_examples.py

```diff
@@ -43,3 +45,11 @@ class ExamplesTests(TestCase):
                            glob.glob(os.path.join(example_dir(), '*.py'))))
     def test_example(self, name, example):
         imp.load_source('__main__', os.path.basename(example), open(example))
+        # The algorithms buyapple.py and dual_moving_average.py have no
+        # if __name__ == '__main__' code block, and consequently are not
+        # executed by the imp.load_source() call. Instead, call these
+        # example algorithms directly as if using scripts/run_algo.py
+        if name == 'buyapple_py' or name == 'dual_moving_average_py':
+            confs = ['-f', example, '--start', '2011-1-1', '--end', '2012-1-1']
+            parsed_args = parse_args(confs)
+            run_pipeline(**parsed_args)
```

Comment:
@twiecki  I realized that to date we have not actually been calling the buyapple.py or dual_moving_average.py examples algorithms, as they have no `if __name__ == ‘__main__’` code block. The above if-statement addresses this. The alternative was to include an `if __name__ == ‘__main__’` block in each of these algorithms, but I wanted to avoid this as it would introduce yet more differences between pure zipline code and that run on quantopian.com. This update will ensure we have test coverage for these two algorithms, and their corresponding analyze functions. The absence of this coverage resulted in the original user-reported error https://github.com/quantopian/zipline/issues/663.

What do you think about going one step further and removing the `if __name__ == ‘__main__’` from the other example algorithms? The advantage of this would be that there would only be only way of executing an algorithm from the command line, using `python scripts/run_algo.py`, which would clear up some user confusion, and users could more easily copy-paste into quantopian.com.